test/Azure.Functions.Cli.Tests/E2E/DeploymentTests.cs

```diff
@@ -21,19 +21,42 @@ public class DeploymentTests : BaseE2ETest, IClassFixture<StorageAccountManager>
         private readonly static string linuxStorageAccountName = $"lcte2estorage{id}";
         private readonly static string linuxConsumptionServerFarm = $"lcte2ecserverfarm{id}";
         private readonly static string linuxConsumptionPythonFunctionApp = $"lcte2ecpython{id}";
+        private bool ResroucesCreated = false;
 
         public DeploymentTests(ITestOutputHelper output, StorageAccountManager saManager, ServerFarmManager sfManager, FunctionAppManager faManager) : base(output)
         {
             _storageAccountManager = saManager;
             _serverFarmManager = sfManager;
             _functionAppManager = faManager;
-            if ((!_storageAccountManager.Contains(linuxStorageAccountName) ||
+
+            if (EnvironmentHelper.GetEnvironmentVariableAsBool(E2ETestConstants.IsPublicBuild) == true)
+            {
+                // no need to create the resources for public build.
+                return;
+            }
+
+            if (!EnvironmentHelper.GetEnvironmentVariableAsBool(E2ETestConstants.EnableDeploymentTests))
+            {
+                // no need to create the resources if deployment tests are not enabled.
+                return;
+            }
```

Comment:
Do these need to be added here? Can't you just have the steps skipped in the yml file?

Reply:
The environment comes from yml file. Since the same code is shared between official/public therefore this check ensure that public doesn't execute it.